game-core/src/main/java/games/strategy/engine/data/UnitType.java

```diff
@@ -72,13 +72,21 @@ public int hashCode() {
     return Objects.hashCode(getName());
   }
 
+  /**
+   * Get unit type tooltip checking for custom tooltip content.
+   */
   public String getTooltip(final PlayerID playerId) {
     final String customTip = TooltipProperties.getInstance().getToolTip(this, playerId);
-    if (customTip == null || customTip.trim().length() <= 0) {
-      return UnitAttachment.get(this).toStringShortAndOnlyImportantDifferences(
-          (playerId == null ? PlayerID.NULL_PLAYERID : playerId));
+    if (customTip != null && customTip.trim().length() > 0) {
+      return LocalizeHtml.localizeImgLinksInHtml(customTip.trim());
+    }
+    final String generated = UnitAttachment.get(this)
+        .toStringShortAndOnlyImportantDifferences((playerId == null ? PlayerID.NULL_PLAYERID : playerId));
+    final String appendedTip = TooltipProperties.getInstance().getAppendedToolTip(this, playerId);
+    if (appendedTip != null && appendedTip.trim().length() > 0) {
```

Comment:
nit, can `appendedTip` be null? It looks like `getAppendedToolTip` would guarantee a non-null return type. And then to avoid IDE complaints,`appendedTip.trim().length() > 0` -> `!appendedTip.isEmpty()`  (the trim looks unnecesary, but if needed it seems like it could/should be attached to `getAppendedToolTip`)

Reply:
Yeah, appears it can't be null so updated this and the one above it.